src/client/components/app/Transactions/ClaimRevenueTx.tsx

```diff
@@ -0,0 +1,348 @@
+          />
+        );
+      }
+      if (input.type === 'bytes') {
+        inputFieldsHTML.push(
+          <TxByteInput
+            headerText={input.name}
+            byteCode={userFuncInputs[input.name] ?? ''}
+            onByteCodeChange={(e: any) => handleInputChange(input.name, input.type, e)}
+          />
+        );
+      }
+    }
+
+    const mergedInputFieldsHTML = (
+      <React.Fragment>
+        <Header>{t('components.transaction.claim-revenue.rev-contract-inputs')}</Header>
+        {inputFieldsHTML}
+      </React.Fragment>
+    );
+    return mergedInputFieldsHTML;
+  };
+
+  // TODO: Fix logic to autopopulate claim revenue function within the modal.
+  // Logic to successfully call claim revenue within the modal
+  // console.log('Rev Contract Mapping funcs', selectedContractFunctions);
+  // console.log('Rev Contract Mapping', selectedContractFunctions[selectedRevenueContract]);
+  // const selectedFunc = {
+  //   id: '0',
+  //   label: 'claimPullPayment', //selectedContractFunctions[selectedRevenueContract][actualClaimFunc],
+  //   value: '',
+  // };
+  // const funcOptions2 = [selectedFunc];
+
+  // const funcInputs2 = generateClaimFuncInputs(selectedFunc.label, contractABI[selectedRevenueContract]!);
+  // console.log('Func Inputs: ', funcInputs2);
+  // setFuncInputs(funcInputs2);
+  // const hashedSigFunc = generateSig(selectedFunc.label, contractABI[selectedRevenueContract]!);
+  // console.log('Rev Contract Mapping - hashedSigFunc: ', hashedSigFunc);
+  // setClaimFunc(hashedSigFunc);
+  // setClaimFuncType(selectedFunc);
+
+  return (
+    <StyledTransaction onClose={onClose} header={header}>
+      <TxAddressInput
+        headerText={t('components.transaction.enable-spigot.revenue-contract')}
+        address={selectedRevenueContract}
+        //onAddressChange={onRevContractChange}
+      />
+      <TxTokenInput
+        headerText={t('components.transaction.enable-spigot.revenue-token-to-claim')}
+        selectedToken={selectedSellToken!}
+        onSelectedTokenChange={onSelectedSellTokenChange}
+        tokenOptions={sourceAssetOptions}
+        amount="0"
+        readOnly={true}
+        hideAmount={true}
+      />
+
+      {isVerifiedContract ? (
+        <>
+          <TxFuncSelector
+            headerText={t('components.transaction.enable-spigot.function-revenue')}
+            typeOptions={funcOptions}
+            selectedType={claimFuncType}
+            onSelectedTypeChange={onClaimFuncSelection}
+          />
+        </>
+      ) : (
+        // if no ABI, input bytecode manually
+        <>
+          <TxByteInput
+            headerText={t('components.transaction.enable-spigot.function-revenue')}
+            inputText={' '}
+            inputError={false}
+            byteCode={claimFunc}
+            onByteCodeChange={setClaimFunc}
+            readOnly={false}
+          />
+        </>
+      )}
+      {generateInputFields()}
```

Comment:
shouldnt we only do this if we have a verified contract? So should fit in the ternary operator above?

Reply:
fixed

Reply:
fixed